array/array.mbt

```diff
@@ -92,10 +119,8 @@ test "iteri" {
 /// ```
 pub fn map[T, U](self : Array[T], f : (T) -> U) -> Array[U] {
   let res = Array::make(self.length(), f(self[0]))
-  let mut i = 0
```

Comment:
you need check if the array is empty or not, otherwise this would throw an exception